pkg/kubelet/cm/memorymanager/state/checkpoint.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package state
+
+import (
+	"encoding/json"
+
+	"k8s.io/kubernetes/pkg/kubelet/checkpointmanager"
+	"k8s.io/kubernetes/pkg/kubelet/checkpointmanager/checksum"
+)
+
+var _ checkpointmanager.Checkpoint = &MemoryManagerCheckpoint{}
+
+// MemoryManagerCheckpoint struct is used to store memory/pod assignments in a checkpoint
+type MemoryManagerCheckpoint struct {
+	MachineState MemoryMap                  `json:"machineState"`
+	Entries      ContainerMemoryAssignments `json:"entries,omitempty"`
+	Checksum     checksum.Checksum          `json:"checksum"`
+}
+
+// NewMemoryManagerCheckpoint returns an instance of Checkpoint
+func NewMemoryManagerCheckpoint() *MemoryManagerCheckpoint {
+	//lint:ignore unexported-type-in-api user-facing error message
+	return &MemoryManagerCheckpoint{
+		Entries:      ContainerMemoryAssignments{},
+		MachineState: MemoryMap{},
+	}
+}
+
+// MarshalCheckpoint returns marshalled checkpoint
+func (mp *MemoryManagerCheckpoint) MarshalCheckpoint() ([]byte, error) {
+	// make sure checksum wasn't set before so it doesn't affect output checksum
+	mp.Checksum = 0
```

Comment:
Why `Checksum` has to be zeroed?
Okay, never mind :smile: 
I saw the code comes from: `kubernetes/pkg/kubelet/cm/cpumanager/state/checkpoint.go`

Reply:
It dropped to avoid including checksum into the calculation of the checksum.

Reply:
Okay, it's obvious now. Thanks.